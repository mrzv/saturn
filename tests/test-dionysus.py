# vim: ft=python foldmethod=marker foldlevel=0

import dionysus as d

#---#

simplices = [([2], 4), ([1,2], 5), ([0,2], 6),
             ([0], 1),   ([1], 2), ([0,1], 3)]
f = d.Filtration()
for vertices, time in simplices:
    f.append(d.Simplex(vertices, time))
f.sort()
for s in f:
   print(s)
#var> f

#---#

#var> f

m = d.homology_persistence(f)

#chk>

#---#

dgms = d.init_diagrams(m,f)
print(dgms)
#var> dgms
