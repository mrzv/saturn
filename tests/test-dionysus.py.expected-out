>>> # vim: ft=python foldmethod=marker foldlevel=0                              
...                                                                             
... import dionysus as d                                                        


>>> simplices = [([2], 4), ([1,2], 5), ([0,2], 6),                              
...              ([0], 1),   ([1], 2), ([0,1], 3)]                              
... f = d.Filtration()                                                          
... for vertices, time in simplices:                                            
...     f.append(d.Simplex(vertices, time))                                     
... f.sort()                                                                    
... for s in f:                                                                 
...    print(s)                                                                 

<0> 1
<1> 2
<0,1> 3
<2> 4
<1,2> 5
<0,2> 6
───────────────────── Failed to save varialbes f, skipping ─────────────────────

───────── Warning: isolated variable cell f; result will never be used ─────────
───────────────────── Failed to save variables f, skipping ─────────────────────
>>> m = d.homology_persistence(f)                                               


────────────── Failed to save state in checkpoint cell, skipping ───────────────
>>> dgms = d.init_diagrams(m,f)                                                 
... print(dgms)                                                                 

[Diagram with 3 points, Diagram with 1 points]

