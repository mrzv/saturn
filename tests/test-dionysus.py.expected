# vim: ft=python foldmethod=marker foldlevel=0

import dionysus as d

#---#

simplices = [([2], 4), ([1,2], 5), ([0,2], 6),
             ([0], 1),   ([1], 2), ([0,1], 3)]
f = d.Filtration()
for vertices, time in simplices:
    f.append(d.Simplex(vertices, time))
f.sort()
for s in f:
   print(s)
#var> f
#o> <0> 1
#o> <1> 2
#o> <0,1> 3
#o> <2> 4
#o> <1,2> 5
#o> <0,2> 6

#---#

#var> f

m = d.homology_persistence(f)

#chk>

#---#

dgms = d.init_diagrams(m,f)
print(dgms)
#var> dgms
#var>{{{
#var>gANDILoOz7x2TVxUDPjIDjqo7gKLvtNlFEqEEHWG4BS+BCGicQAugANdcQAoY2Rpb255c3VzLl9kaW9u
#var>eXN1cwpEaWFncmFtCnEBKYFxAl1xAyhHP/AAAAAAAABHf/AAAAAAAABLAIdxBEdAAAAAAAAAAEdACAAA
#var>AAAAAEsBh3EFR0AQAAAAAAAAR0AUAAAAAAAASwOHcQZlhXEHYmgBKYFxCF1xCUdAGAAAAAAAAEd/8AAA
#var>AAAAAEsFh3EKYYVxC2JlLg==
#var>}}}
#o> [Diagram with 3 points, Diagram with 1 points]
